src/sonic-yang-models/yang-models/sonic-events-dhcp-relay.yang

```diff
@@ -40,16 +40,11 @@ module sonic-events-dhcp-relay {
                     name of the interface discarding.";
 
             leaf ifname {
-                type union {
-                    type leafref {
-                        path "/port:sonic-port/port:PORT/port:PORT_LIST/port:name"; 
-                    }
-                    type leafref {
-                        path "/lag:sonic-portchannel/lag:PORTCHANNEL/lag:PORTCHANNEL_LIST/lag:name";
-                    }
+                type string {
+                    pattern 'Ethernet[0-9]{1,2}';
```

Comment:
>Ethernet [](http://example.com/codeflow?start=29&length=8)

There is use case like `Ethernet124` not covered by this pattern.
PortChannel may be a valid use case if there is PortChannel in Vlan.

Reply:
Changed pattern.

I checked with Kelly, for this event it should just be Ethernets, PortChannel is used for sending out packets

Reply:
Could you explain more about the disadvantage of using leafref PORT_LIST? The pattern string is not future proof.

Reply:
I initially had it as a leafref, but we can't use it for our use case, because we are getting events as a json format such as 

              "stdout_lines": [
                   "2023-01-10 01:02:45,053 [INFO] CURRENT JSON_DATA: {\"sonic-events-swss:sonic-events-swss\":{", 
                   "  \"sonic-events-swss:if-state\": {", 
                   "    \"ifname\": \"Ethernet8\",", 
                   "    \"status\": \"down\",", 
                   "    \"timestamp\": \"2023-01-10T01:02:30.349203Z\"", 
                   "  }", 
                   "}}", 

In order to use a leafref I would need a reference such as the following as part of my json, which is not possible 
      "sonic-port:sonic-port": {
            "sonic-port:PORT": {
                "PORT_LIST": [
                    {
                        "admin_status": "up",
                        "alias": "eth0",
                        "description": "Ethernet0",
                        "lanes": 65,
                        "mtu": 9000,
                        "name": "Ethernet0",
                        "speed": 25000
                    }
                ]
            }
        }, 

"2023-01-10 01:02:45,056 [INFO] Exception thrown: Leafref \"/sonic-port:sonic-port/sonic-port:PORT/sonic-port:PORT_LIST/sonic-port:name\" of value \"Ethernet8\" points to a non-existing leaf." 